Lib/test/test_sort.py

```diff
@@ -373,6 +373,10 @@ def test_unsafe_tuple_compare(self):
         check_against_PyObject_RichCompareBool(self, [float('nan')]*100)
         check_against_PyObject_RichCompareBool(self, [float('nan') for
                                                       _ in range(100)])
+
+    def test_not_all_tuples(self):
+        self.assertRaises(TypeError, [(1.0, 1.0), (False, "A"), 6].sort)
+        self.assertRaises(TypeError, [('a', 1), (1, 'a')].sort)
```

Comment:
Please add also the test for the case when the first key is int and the second is not int: `[(1, 'a'), ('a', 1)]`.